zod/app/routes/products.tsx

```diff
@@ -0,0 +1,198 @@
+import { LoaderArgs, json, redirect } from "@remix-run/node";
+import {
+  Form,
+  Link,
+  useLoaderData,
+  useNavigation,
+  useSubmit,
+} from "@remix-run/react";
+import * as z from "zod";
+
+import { mockProducts } from "~/lib/product.server";
+
+export const loader = ({ request }: LoaderArgs) => {
+  const URLSearchParams = new URL(request.url).searchParams;
```

Comment:
Let's write a little less code here

```suggestion
  const { searchParams } = new URL(request.url);
```